tests/ui/impl-trait/can-return-unconstrained-closure.rs

```diff
@@ -1,23 +0,0 @@
-// Test that we are special casing "outlives" for opaque types.
-//
-// The return type of a closure is not required to outlive the closure. As such
-// the following code would not compile if we used a standard outlives check
-// when checking the return type, because the return type of the closure would
-// be `&ReEmpty i32`, and we don't allow `ReEmpty` to occur in the concrete
-// type used for an opaque type.
-//
-// However, opaque types are special cased to include check all regions in the
-// concrete type against the bound, which forces the return type to be
-// `&'static i32` here.
```

Comment:
we're not special-casing anything here :3 unless "require member regions to be equal to one of the arg regions" is meant by that. Given that we've got a lot of test which return closures, this is redundant